### ISSUE TITLE
Versions of ROS, ROSPlan, and mlconjug3?

### ISSUE THREAD

**Issue by adaruna3** [gerardcanal/task_plan_verbalization#1]
Could you provide the versions of the dependencies in the issue title? 

Specifically, the latest mlconjug3 no longer supports Python 2.X. I am assuming this ROS package was developed for ROS melodic given dependencies of ROSPlan, so there's probably a specific version of mlconjug3 to use.

I am looking through mlconjug3 commits to see when Python 2.X support was dropped but it would be helpful to have the right version.

If installed with pip, an output of `pip show mlconjug3` would be great.

**Comment by gerardcanal** [comment]
Hi Angel,
PlanVerb is developed in Python 3, and works with mlconjug3. I used the version in ROSPlan's master which works for ROS Melodic. I did some PRs in ROSPlan which improved some things but not all of them have been yet merged, it should work well nonetheless. I have not yet tried in noetic but will do it soon. 

Will make this now clearer on the readme!

The current version I have installed for mlconjug3 is the following:
```
$ pip show mlconjug3
Name: mlconjug3
Version: 3.7.13
Summary: A Python library to conjugate French, English, Spanish, Italian, Portuguese and Romanian verbs using Machine Learning techniques.
Home-page: https://github.com/SekouDiaoNlp/mlconjug3
Author: SekouDiaoNlp
Author-email: diao.sekou.nlp@gmail.com
License: MIT license
Location: /home/gerard/.local/lib/python3.6/site-packages
Requires: Click, colorama, cython, defusedxml, joblib, numpy, scikit-learn, scipy
Required-by: 
```

**Comment by adaruna3** [comment]
Hi Gerard,

Got it. I will be switching to Python3 with Melodic then to build/run this package.

Thanks!
Angel

**Comment by gerardcanal** [comment]
You're welcome!
I added in the readme the packages I had to install to run Python3 in ROS Melodic. I looked for them now so it might not be complete, please let me know if there's anything else needed to run it!

**Comment by adaruna3** [comment]
Perfect! Yes, seems like some magic is needed to get Python3 in ROS Melodic, so thanks for the pointers!

I will update you if anything comes up to add to docs. Thanks!!

**Comment by adaruna3** [comment]
Update: Works great! After getting the right python version install was smooth. A few additional packages were needed but the fixes are very obvious from the error output (e.g., no module 'spacy'). Thanks again!

**Comment by gerardcanal** [comment]
Great! My bad about Spacy, that was a later addition (fixing the README now).

Thanks for the update!

**Comment by adaruna3** [comment]
No worries at all. I just did a `history | grep pip3` to see what packages were added. Spacy is one, another is empy, and last is nltk.

Adding a `pip3 install spacy empy nltk` followed by `python3 -m spacy download en_core_web_sm` should get everything. I am not sure if those were dependencies of the other packages or not, but those were the only pip packages that were missing for me.

And for others that come here to troubleshoot installation, I didn't do anything to get Python3 working with my original ROS Melodic installation other than creating an anaconda environment using Python3 followed by running the command Gerard added to the README.
